File: src/main/java/fr/uga/l3miage/pc/prisonersdilemma/dto/InitPartieDTO.java
package fr.uga.l3miage.pc.prisonersdilemma.dto;

import fr.uga.l3miage.pc.prisonersdilemma.enums.TypeStrategie;
import lombok.Getter;
import lombok.Setter;
import jakarta.validation.constraints.*;



@Getter
@Setter

public class InitPartieDTO {

    @NotNull(message = "Le nom d'utilisateur est obligatoire.")
    @Size(min = 3, max = 20, message = "Le nom d'utilisateur doit comporter entre 3 et 20 caractères.")
    private String nomJoueur1;
    @Size(min = 3, max = 20, message = "Le nom d'utilisateur doit comporter entre 3 et 20 caractères.")
    @NotNull(message = "Le nom d'utilisateur est obligatoire.")
    private String nomJoueur2;
    @NotNull
    private boolean joueur1bot;
    @NotNull
    private boolean joueur2bot;
    private TypeStrategie strategieJoueur1;

    private boolean strategieExterneJoueur1=false;
    private boolean strategieExterneJoueur2=false;

    private TypeStrategie strategieJoueur2;
    @Positive
    private int nbMaxTours;
}

File: src/main/front-dilemme/src/app/models/init-partie-dto.ts
import { TypeStrategie } from './type-strategie';

export interface InitPartieDTO {
    nomJoueur1: string;
    nomJoueur2: string;
    joueur1bot: boolean;
    joueur2bot: boolean;
    strategieJoueur1: TypeStrategie | undefined;
    strategieJoueur2: TypeStrategie | undefined ;
    strategieExterneJoueur1: boolean;
    strategieExterneJoueur2: boolean;
    nbMaxTours: number;
  }
